library/modules/EventManager.cpp

```diff
@@ -235,6 +239,9 @@ static int32_t reportToRelevantUnitsTime = -1;
 //interaction
 static int32_t lastReportInteraction;
 
+//unit action
+static std::unordered_map<int32_t,std::vector<int32_t> > unitToKnownActions;
```

Comment:
It looks like this should be a `std::unordered_map<int32_t,std::unordered_set<int32_t>>` since there doesn't seem to be a reason to pay the cost of iterating through the vector on every lookup, or am I missing something?

Reply:
Your map<set> is a fine approach, but I still think using the action event data alone would be best. `std::unordered_set<ActionData>` with a hash function that hashes on the unit id and the action id. Maybe I'm missing something, but that looks like it'll give the same guarantees with the fastest lookup times

Reply:
Agreed. That would be cleaner.

Reply:
@Putnam3145 are you ok with this change?